build-test.sh

```diff
@@ -727,6 +735,24 @@ while :; do
             __ClangMinorVersion=0
             ;;
 
+        gcc5|-gcc5)
+            __GccMajorVersion=5
+            __GccMinorVersion=
+            __GccBuild=1
+            ;;
+
+        gcc7|-gcc7)
```

Comment:
gcc6 and gcc8 (current) could also be added. In CoreFX, I tested gcc5 to gcc8. Quickest way was to use the official gcc dockers: https://hub.docker.com/_/gcc/ (also provides gcc5 to gcc8 images)